dbt_platform_helper/domain/update_alb_rules.py

```diff
@@ -0,0 +1,338 @@
+                    )
+                    rule_arn = self.load_balancer.create_rule(
+                        listener_arn,
+                        rule["actions"],
+                        rule["conditions"],
+                        rule_priority,
+                        tags=[
+                            {"Key": "application", "Value": application_name},
+                            {"Key": "environment", "Value": environment},
+                            {"Key": "service", "Value": rule["service_name"]},
+                            {"Key": "reason", "Value": "service"},
+                            {"Key": "managed-by", "Value": MANAGED_BY_PLATFORM},
+                        ],
+                    )["Rules"][0]["RuleArn"]
+                    operation_state.created_rules.append(rule_arn)
+                    rule_priority += RULE_PRIORITY_INCREMENT
+
+                next_thousandth = lambda x: ((x // 1000) + 1) * 1000
+                rule_priority = next_thousandth(rule_priority)
+
+        if (
+            service_deployment_mode == Deployment.COPILOT.value
+            or service_deployment_mode == Deployment.DUAL_DEPLOY_COPILOT.value
+        ):
+            self._delete_rules(mapped_rules.get(RuleType.PLATFORM.value, []), operation_state)
+
+    def _delete_rules(self, rules: List[dict], operation_state: OperationState):
+        for rule in rules:
+            rule_arn = rule["RuleArn"]
+            try:
+                self.load_balancer.delete_listener_rule_by_resource_arn(rule_arn)
+                operation_state.deleted_rules.append(rule)
+                self.io.debug(f"Deleted existing rule: {rule_arn}")
+            except Exception as e:
+                self.io.error(f"Failed to delete existing rule {rule_arn}: {str(e)}")
+                raise
+
+    def _filter_rule_type(self, rule: dict) -> str:
+        if rule["Tags"]:
+            if rule["Tags"].get("managed-by", "") == MANAGED_BY_PLATFORM:
+                return RuleType.PLATFORM.value
+            if rule["Tags"].get("name", "") in MAINTENANCE_PAGE_TAGS:
+                return RuleType.MAINTENANCE.value
+
+        if rule["Priority"] == "default":
+            return RuleType.DEFAULT.value
+        if int(rule["Priority"]) >= COPILOT_RULE_PRIORITY:
+            return RuleType.COPILOT.value
+
+        return RuleType.MANUAL.value
+
+    def _get_service_from_tg(self, rule: dict) -> str:
+        target_group_arn = ""
+
+        for action in rule["Actions"]:
+            if action["Type"] == "forward":
+                target_group_arn = action["TargetGroupArn"]
+
+        if target_group_arn:
+            tgs = self.load_balancer.get_target_groups_with_tags([target_group_arn])
+
+            # TODO make more robust?
+            for tg in tgs:
+                return tg["Tags"].get("copilot-service", "")
+        else:
+            # TODO add error handling if no target_group_arn
+            self.io.warn("no target group found")
+        return None
```

Comment:
Do a bit of tidying here, make sure we either returning a service name or `None`.